src/adc.rs

```diff
@@ -1,18 +1,75 @@
+///! RX-only interface, it is configured in full-duplex mode and the TX pin is left disconnected.
+///! This allows the SPI interface to periodically read a single word whenever a word is written to
+///! the TX side. Thus, by running a continuous DMA transfer to periodically write a value into the
+///! TX FIFO, we can schedule the regular collection of ADC samples in the SPI RX buffer.
+///!
+///! In order to collect the acquired ADC samples into a RAM buffer, a second DMA transfer is
+///! configured to read from the SPI RX FIFO into RAM. The request for this transfer is connected to
+///! the SPI RX data signal, so the SPI peripheral will request to move data into RAM whenever it is
+///! available. When enough samples have been collected, a transfer-complete interrupt is generated
+///! and the ADC samples are available for processing.
+///!
+///! The SPI peripheral internally has an 8- or 16-byte TX and RX FIFO, which corresponds to a 4- or
+///! 8-sample buffer for incoming ADC samples. During the handling of the DMA transfer completion,
+///! there is a small window where buffers are swapped over where it's possible that a sample could
+///! be lost. In order to avoid this, the SPI RX FIFO is effectively used as a "sample overflow"
+///! region and can buffer a number of samples until the next DMA transfer is configured. If a DMA
+///! transfer is still not set in time, the SPI peripheral will generate an input-overrun interrupt.
+///! This interrupt then serves as a means of detecting if samples have been lost, which will occur
+///! whenever data processing takes longer than the collection period.
+///!
+///!
+///! ## Starting Data Collection
+///!
+///! Because the DMA data collection is automated via timer count comparisons and DMA transfers, the
+///! ADCs can be initialized and configured, but will not begin sampling the external ADCs until the
+///! sampling timer is enabled. As such, the sampling timer should be enabled after all
+///! initialization has completed and immediately before the embedded processing loop begins.
+///!
+///!
+///! ## Batch Sizing
+///!
+///! The ADCs collect a group of N samples, which is referred to as a batch. The size of the batch
+///! is configured by the user at compile-time to allow for a custom-tailored implementation. Larger
+///! batch sizes generally provide for more processing time per sample, but come at the expense of
+///! increased input -> output latency.
+///!
+///!
+///! # Note
+///!
+///! While there are two ADCs, only a single ADC is configured to generate transfer-complete
+///! interrupts. This is done because it is assumed that the ADCs will always be sampled
+///! simultaneously. If only a single ADC is used, it must always be ADC0, as ADC1 will not generate
+///! transfer-complete interrupts.
+///!
+///! There is a very small amount of latency between sampling of ADCs due to bus matrix priority. As
```

Comment:
How might we be able to calculate this latency?

Reply:
The latency is 1 bus matrix clock. However, since the bus might be used by for example the CPU, it might get delayed further. Because this latency is in the range of a handful of AXI bus clocks, that's what I used the phrase "a very small amount"

Reply:
Essentially, the latency is not deterministic, hence why it's not clearly specified here

Reply:
Could we specify some sort of worst case number? This allows someone to make the determination of "I don't care about this latency".